include/circt/Dialect/Seq/Seq.td

```diff
@@ -47,10 +50,31 @@ def CompRegOp : SeqOp<"compreg",
     Optional<I1>:$reset, Optional<AnyType>:$resetValue);
   let results = (outs AnyType:$data);
 
-  let printer = "return ::print(p, *this);";
+  let printer = "return ::print$cppClass(p, *this);";
```

Comment:
This should not be necessary as we can (and do) rely on C++ overloading.

Reply:
FWIW, we aren't consistent about this, but I personally prefer the `print$cppClass` - the error case is much easier to debug and reason about, because the C++ compiler says "hey I don't have `printFoo` instead of spew out 100 `print`s that aren't the right match.

Reply:
Yeah, that's a good point. I haven't seen any other place in the code where we use `printFoo`, but maybe we should move over. Consistency is good.

Reply:
#1412